### PR TITLE
audio: dai-zephyr: convert DMA src/dst addresses to host-relative addresses

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
The DMAC has the same view of the address space as the HOST. As such, the source and destination addresses need to be converted to HOST-relative addresses before attempting to perform a DMA transfer. This is only relevant for platforms for which the DSP and HOST have different views of the address space (e.g: i.MX8ULP).

### CODE REVIEW

File: src/audio/dai-zephyr.c [hunk 1]
```diff
@@ -751,8 +751,17 @@ static int dai_set_dma_config(struct dai_data *dd, struct comp_dev *dev)
 	for (i = 0; i < dma_cfg->block_count; i++) {
 		dma_block_cfg->dest_scatter_en = config->scatter;
 		dma_block_cfg->block_size = config->elem_array.elems[i].size;
-		dma_block_cfg->source_address = config->elem_array.elems[i].src;
-		dma_block_cfg->dest_address = config->elem_array.elems[i].dest;
+		if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
```
Comment (iuliana-prodan) [inline src/audio/dai-zephyr.c:754]: Why add a new `if` based on direction? 
You can use the one below, from line [765](https://github.com/thesofproject/sof/pull/8948/files#diff-cf0363219538c1e144ce83bdf7ba8aef8c141f26ffe6e2dafff869623360875bR765).

Comment (LaurentiuM1234) [reply src/audio/dai-zephyr.c:762]: my bad, fixed!

**Review by dbaluta — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
@wszypelt Can you check? Job 13725213 was cancelled and exceptation is this PR is a no-op for Intel platforms.

**Comment by wszypelt** [comment]
@kv2019i Internal Intel CI System - all green

**Pull request merged**